modules/mapper-extras/src/internalClusterTest/java/org/elasticsearch/index/mapper/RankFeaturesMapperIntegrationIT.java

```diff
@@ -0,0 +1,116 @@
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.mapper.extras.MapperExtrasPlugin;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.test.ESIntegTestCase;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Map;
+
+import static org.elasticsearch.xcontent.XContentFactory.jsonBuilder;
+import static org.hamcrest.Matchers.equalTo;
+
+public class RankFeaturesMapperIntegrationIT extends ESIntegTestCase {
+
+    private static final String LOWER_RANKED_FEATURE = "ten";
+    private static final String HIGHER_RANKED_FEATURE = "twenty";
+    private static final String INDEX_NAME = "rank_feature_test";
+    private static final String FIELD_NAME = "all_rank_features";
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return Arrays.asList(MapperExtrasPlugin.class);
+    }
+
+    public void testRankFeaturesTermQuery() throws IOException {
+        init();
+        SearchResponse response = client().prepareSearch(INDEX_NAME)
+            .setQuery(QueryBuilders.termQuery(FIELD_NAME, HIGHER_RANKED_FEATURE))
+            .get();
+        assertThat(response.getHits().getTotalHits().value, equalTo(2L));
+        for (SearchHit hit : response.getHits().getHits()) {
+            assertThat(hit.getScore(), equalTo(20f));
+        }
+
+        response = client().prepareSearch(INDEX_NAME)
+            .setQuery(QueryBuilders.termQuery(FIELD_NAME, HIGHER_RANKED_FEATURE).boost(100f))
+            .get();
+        assertThat(response.getHits().getTotalHits().value, equalTo(2L));
+        for (SearchHit hit : response.getHits().getHits()) {
+            assertThat(hit.getScore(), equalTo(2000f));
+        }
+
+        response = client().prepareSearch(INDEX_NAME)
+            .setQuery(
+                QueryBuilders.boolQuery()
+                    .should(QueryBuilders.termQuery(FIELD_NAME, HIGHER_RANKED_FEATURE))
+                    .should(QueryBuilders.termQuery(FIELD_NAME, LOWER_RANKED_FEATURE))
+                    .minimumShouldMatch(1)
+            )
+            .get();
+        assertThat(response.getHits().getTotalHits().value, equalTo(3L));
+        for (SearchHit hit : response.getHits().getHits()) {
+            if (hit.getId().equals("all")) {
+                assertThat(hit.getScore(), equalTo(30f));
+            }
+            if (hit.getId().equals("lower")) {
+                assertThat(hit.getScore(), equalTo(10f));
+            }
+            if (hit.getId().equals("higher")) {
+                assertThat(hit.getScore(), equalTo(20f));
+            }
+        }
+
+        response = client().prepareSearch(INDEX_NAME).setQuery(QueryBuilders.termQuery(FIELD_NAME, "missing_feature")).get();
+        assertThat(response.getHits().getTotalHits().value, equalTo(0L));
```

Comment:
Is it worth testing boosted queries too to make sure the query boost is applied?

Reply:
Higher up in the test I have a `term` query that is `boost(100f)` to test the limit and verify scoring is correct.